src/main/java/moze_intel/projecte/gameObjs/items/rings/HarvestGoddess.java

```diff
@@ -52,25 +53,21 @@ public void onUpdate(ItemStack stack, World world, Entity entity, int par4, bool
 		super.onUpdate(stack, world, entity, par4, par5);
 		
 		EntityPlayer player = (EntityPlayer) entity;
-		
-		if (ItemHelper.getOrCreateCompound(stack).getBoolean(TAG_ACTIVE))
-		{
-			double storedEmc = getEmc(stack);
-			
-			if (storedEmc == 0 && !consumeFuel(player, stack, 64, true))
-			{
-				stack.getTagCompound().setBoolean(TAG_ACTIVE, false);
-			}
-			else
-			{
-				WorldHelper.growNearbyRandomly(true, world, new BlockPos(player), player);
-				removeEmc(stack, 0.32F);
+		if (player.getCapability(InternalTimers.CAPABILITY, null).canRingUpdate()) {
+
+			if (ItemHelper.getOrCreateCompound(stack).getBoolean(TAG_ACTIVE)) {
+				long storedEmc = getEmc(stack);
+
+				if (storedEmc == 0 && !consumeFuel(player, stack, 64, true)) {
+					stack.getTagCompound().setBoolean(TAG_ACTIVE, false);
+				} else {
+					WorldHelper.growNearbyRandomly(true, world, new BlockPos(player), player);
+					removeEmc(stack, 6);
```

Comment:
6 emc every n ticks does not add up to 0.32 emc per tick?

Reply:
Was one of the many examples of how it doesn't translate well. I need a better solution to all items, not just this one